ceph-perf-pull-requests/config/definitions/ceph-perf-pull-requests.yml

```diff
@@ -0,0 +1,130 @@
+# macros
+
+- scm:
+    name: ceph-master
+    scm:
+      - git:
+          url: https://github.com/ceph/ceph.git
+          branches:
+            - origin/master
+          skip-tag: true
+          timeout: 20
+          basedir: "ceph-master"
+          shallow-clone: true
+          wipe-workspace: false
+
+- scm:
+    name: ceph-pr
+    scm:
+      - git:
+          url: https://github.com/ceph/ceph.git
+          branches:
+            - ${sha1}
+          refspec: +refs/pull/${ghprbPullId}/*:refs/remotes/origin/pr/${ghprbPullId}/*
+          timeout: 20
+          basedir: "ceph-pr"
+          shallow-clone: true
+          wipe-workspace: true
+
+- scm:
+    name: cbt
+    scm:
+      - git:
+          url: https://github.com/tchaikov/cbt.git
+          branches:
+            - origin/wip-requirement.txt
+          timeout: 20
+          wipe-workspace: false
+          basedir: "cbt"
+          shallow-clone: true
+
+- builder:
+    name: run-cbt
+    builders:
+      - shell: |
+          export NPROC=$(nproc)
+          cd ceph-{pr_or_master}
+          timeout 7200 src/script/run-make.sh --cmake-args "-DWITH_SEASTAR=ON -DWITH_PYTHON3=ON -DWITH_PYTHON2=OFF -DMGR_PYTHON_VERSION=3.5 -DWITH_SYSTEM_BOOST=ON -DWITH_TESTS=OFF" vstart-base crimson-osd
+          src/script/run-cbt.sh --build-dir build --source-dir .  --cbt ${{WORKSPACE}}/cbt -a ${{WORKSPACE}}/cbt-results/{pr_or_master} src/test/crimson/cbt/radosbench_4K_read.yaml
+
+- builder:
+    name: compare-cbt-results
+    builders:
+      - shell: |
+          ${{WORKSPACE}}/cbt/compare.py -a ${{WORKSPACE}}/cbt-results/{pr} -b ${{WORKSPACE}}/cbt-results/{master} -v
+
+- job:
+    name: ceph-perf-pull-requests
+    project-type: freestyle
+    defaults: global
+    concurrent: true
+    node: huge && (centos7 || trusty) && rebootable
```

Comment:
performance tests need specific machines, these machines require a label so that these tests can identify them. I recently added one manually. Some coordination needs to happen so that this works with labels that are expected.

Reply:
@alfredodeza yeah, i am still struggling with the glue code to drive cbt. and later will try to use github API to send a table to summarize the test result instead of just a single string.

are you talking about incerta08? i was trying to build ceph on RHEL8, but seems we are not quite there yet.